sound/soc/sof/ipc4-topology.c

```diff
@@ -1318,6 +1359,34 @@ static int sof_ipc4_prepare_mixer_module(struct snd_sof_widget *swidget,
 	return sof_ipc4_widget_assign_instance_id(sdev, swidget);
 }
 
+static int sof_ipc4_prepare_process_module(struct snd_sof_widget *swidget,
+					   struct snd_pcm_hw_params *fe_params,
+					   struct snd_sof_platform_stream_params *platform_params,
+					   struct snd_pcm_hw_params *pipeline_params, int dir)
+{
+	struct snd_soc_component *scomp = swidget->scomp;
+	struct snd_sof_dev *sdev = snd_soc_component_get_drvdata(scomp);
+	struct sof_ipc4_process *process = swidget->private;
+	int ret;
+
+	process->available_fmt.ref_audio_fmt = &process->available_fmt.base_config->audio_fmt;
+
+	/*
+	 * Output format is not a must. Process modules should handle output format itself.
```

Comment:
this comment is not very clear. Do you mean to say that output format is optional?

Reply:
> this comment is not very clear. Do you mean to say that output format is optional?

@ranj063 Yes. Some modules doesn't need output format.